tools/python/migration-scripts/Ingest_S3_to_postgressql/README.md

```diff
@@ -0,0 +1,115 @@
+- **File Management**: Moves processed files to a designated directory so retrying is resuming but not restarting
+- **Comprehensive Logging**: Detailed logs for monitoring and troubleshooting
+- **Error Notifications**: Optional SNS notifications for failures
+- **Secure Credentials**: Retrieve database passwords from AWS Secrets Manager
+
+## Prerequisites
+
+- Python 3.6+
+- Aurora/RDS Postgres Cluster schema, database and table with correct datatypes created.
+- AWS credentials configured if using Secrets Manager and Simple Notification Services
+- Download CSV files generated by Timestream unload tool
+
+
+## Installation
+
+```bash
+pip install boto3 psycopg2-binary
+```
+
+## Usage
+
+```bash
+python copy_postgres.py \
+  --table TABLE_NAME \
+  --database DATABASE_NAME \
+  --csv_files_dir "/path/to/csv/files/*.csv" \
+  --host DATABASE_HOST \
+  --secret_name SECRET_NAME \
+  [--schema SCHEMA] \
+  [--user USERNAME] \
+  [--port PORT] \
+  [--parallel_threads THREADS] \
+  [--processed_dir PROCESSED_DIR] \
+  [--logs_dir LOGS_DIR] \
+  [--sns_topic_arn SNS_TOPIC_ARN]
+```
+
+## Parameters
+
+### Required Parameters
+
+- `--table`: Target PostgreSQL table name
+- `--database`: PostgreSQL database name
+- `--csv_files_dir`: Directory or glob pattern for CSV files
+- `--host`: Aurora/RDS Postgres cluster writer endpoint
+- `--secret_name`: AWS Secrets Manager secret containing database password
+
+### Optional Parameters
+
+- `--schema`: PostgreSQL schema (default: public)
+- `--user`: PostgreSQL username (default: postgres)
+- `--port`: PostgreSQL port (default: 5432)
+- `--parallel_threads`: Number of concurrent threads (default: 10)
+- `--processed_dir`: Directory for processed files (default: auto-generated)
+- `--logs_dir`: Directory for log files (default: ./postgres-ingestion-logs)
+- `--sns_topic_arn`: SNS topic ARN for failure notifications
+
+## How It Works
+
+1. **Initialization**: Parse arguments and set up logging
+2. **Authentication**: Retrieve database credentials from AWS Secrets Manager
+3. **Connection**: Create a connection pool to PostgreSQL
+4. **File Discovery**: Find all CSV files matching the specified pattern
+5. **Parallel Processing**: Distribute files across multiple threads
+6. **Data Loading**: For each file:
+   - Extract column names from the CSV header
+   - Build a dynamic COPY command
+   - Load data with retry logic for transient errors
+   - Move successfully processed files to the processed directory
+7. **Notification**: Send SNS notifications for any failures
+8. **Completion**: Log summary statistics
+
+## Example
+
+```bash
+python copy_postgres.py \
+  --table customers \
+  --database sales \
+  --csv_files_dir "/data/csv_files/*partition*/*.csv" \
+  --host database-1.cluster-123456789012.us-east-1.rds.amazonaws.com \
+  --secret_name  'arn:aws:secretsmanager:us-east-1:123456789012:secret:rds!cluster-xxxxx-xx-xx-xx-xxxxxxxx-xxxxx' \
```

Comment:
The mismatched use of single and double quotes is important here because of the `!cluster` in the secret name causing a zsh error. We need to note this in the documentation.

Reply:
Also this is an ARN , not a secret name.